db/external_sst_file_test.cc

```diff
@@ -2462,13 +2462,40 @@ TEST_P(ExternalSSTFileTest, IngestBehind) {
     ASSERT_OK(Put(Key(i), "memtable"));
     true_data[Key(i)] = "memtable";
   }
+
+  ASSERT_OK(Delete(Key(7)));
+  ASSERT_OK(SingleDelete(Key(8)));
   ASSERT_OK(db_->CompactRange(CompactRangeOptions(), nullptr, nullptr));
+  ASSERT_EQ(Get(Key(8)), "NOT_FOUND");
+  ASSERT_EQ(Get(Key(7)), "NOT_FOUND");
```

Comment:
nit: might be helpful to add some comment to clarify the intention of Delete(Key(7)) and SingleDelete(Key(8)) like "// Test that SingleDelte overwritten by Put is not dropped" though the verification happens further down in the test. 

I was slightly confused and mistakenly thought the Delete and SingleDelete were here to delete some **existing** keys and you verified this by ASSERT_EQ(Get(Key(8)), "NOT_FOUND");  and ASSERT_EQ(Get(Key(7)), "NOT_FOUND");